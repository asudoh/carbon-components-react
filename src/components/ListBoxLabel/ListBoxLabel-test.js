import React from 'react';
import { shallow } from 'enzyme';
import ListBoxLabel from '../ListBoxLabel';

describe('ListBoxLabel', () => {
  it('should render', () => {
    const wrapper = shallow(
      <ListBoxLabel label="Label" count={5} clearSelection={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
