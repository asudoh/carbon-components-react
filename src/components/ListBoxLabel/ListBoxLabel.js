import React from 'react';
import PropTypes from 'prop-types';
import ListBoxBadge from '../ListBoxBadge';

const ListBoxLabel = ({ label, count, clearSelection }) => (
  <div className="bx--list-box__label">
    <ListBoxBadge count={count} clearSelection={clearSelection} />
    {label}
  </div>
);

ListBoxLabel.propTypes = {
  label: PropTypes.node.isRequired,
  count: PropTypes.number.isRequired,
  clearSelection: PropTypes.func.isRequired,
};

export default ListBoxLabel;
