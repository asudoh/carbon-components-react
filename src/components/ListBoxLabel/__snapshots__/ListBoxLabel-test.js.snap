// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxLabel should render 1`] = `
<div
  className="bx--list-box__label"
>
  <ListBoxBadge
    clearSelection={[Function]}
    count={5}
  />
  Label
</div>
`;
